### PR TITLE
feat: add support for request params in mock client

### PR DESCRIPTION
**Pull request by kibertoad**
refs #1448

### CODE REVIEW

**Review by mcollina — CHANGES_REQUESTED** [review]
good work! Docs are missing!

File: types/mock-interceptor.d.ts [hunk 1]
```diff
@@ -50,6 +50,8 @@ declare namespace MockInterceptor {
     body?: string | RegExp | ((body: string) => boolean);
     /** Headers to intercept on. */
     headers?: Record<string, string | RegExp | ((body: string) => boolean)> | ((headers: Record<string, string>) => boolean);
+    /** Query params to intercept on */
+    query?: Record<string, any>;
```
Comment (mcollina) [inline types/mock-interceptor.d.ts:54]: please add a test for this type

Comment (kibertoad) [reply types/mock-interceptor.d.ts:54]: Done!

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by kibertoad** [comment]
@mcollina Thank you! Would it be feasible to release a new version next week? New query param functionality is not very compatible with tests without this piece.

**Pull request merged**